### PR TITLE
Air Chemicals 2: "ATMOS PUMPING CLF3 INTO THE AIR AGAIN"

### PR DESCRIPTION
**Pull request by Iamgoofball**
qjov26g
*poo

now with less lag thanks to god knows what change caused it to get fixed

thanks @sawu-tg for the code

### CODE REVIEW

**Comment by xxalpha** [comment]
How do you turn the chemicals into gas?

**Comment by Iamgoofball** [comment]
@xxalpha 

for canisters:

atmos techs have Industrial Chemical Injectors in their lockers

fill it up with chems, set it to inject, and then click on a canister

open canister up all the way

have clf3 gas will cause 1st degree burns

for pumping into vents:

shut vent off

get beaker(s) of shit

dump beaker(s) of shit into the offline vent

turn vent back on

have clf3 gas will cause 1st degree burns

bonus:
don't shut off vent, try and dump in, get it shot back out at you instantly

**Comment by Saegrimr** [comment]
So is it just burny chlorine shit or can we put healing chems in there too? Or start pumping crank into viro's office?

**Comment by Iamgoofball** [comment]
@Saegrimr literally any chemical can be put into the airstream

polonium meth

initro cyanide

styptic powder silver sulf synthflesh salbutamol

any chemical

**Comment by Saegrimr** [comment]
I foresee some problems in the future with dark matter air / similar shit. Either way good work.

**Comment by babinsquared** [comment]
maybe change the "you locate a seal" text to just "you inject xxx into the canister's injection port" or something like that because locating the same thing over and over sounds silly

can't wait to pump people full of black powder and crank up the heat

**Comment by Supermichael777** [comment]
^ the moment black powder actual explodes instead of saying it would have exploded

**Comment by GunHog** [comment]
Is there a way to get more injectors?

**Comment by Iamgoofball** [comment]
@GunHog not atm might add an R&D one

**Comment by vista-** [comment]
How are explosive chems handled? When I last tested sawu's thing, all it did was spam the chat with the exploding message and cause severe lag.

**Comment by Iamgoofball** [comment]
@vista- it's noreact, it can't explode

**Comment by Lumipharon** [comment]
Literally filling the air with AIDS, every time as traitor viro. heuheuheuh.

Does the 'turn you into a slime' chem work with this?

**Comment by Iamgoofball** [comment]
@Lumipharon yes but you'd need insane quantities of the chemical to make it worth it

and acid air does not melt, it just transfers into the mob

**Comment by GunHog** [comment]
Can I inject large amounts of chem into a canister, hook the canister to a port, connect the port to the distro pipes, thus poisoning the station?

**Comment by Iamgoofball** [comment]
yes

**Comment by GunHog** [comment]
Have you given thought to how atmos might filter the chems from the air? Also, I injected 100 units of acid into an oxygen canister and released it. It shot the CPU to 60 and began creating lag. Sadly, gaseous acid is very disappointing, so stick to smoke I guess.

**Comment by Iamgoofball** [comment]
@GunHog What? I was getting extremely minimal lag on my end from it.

**Comment by Midaychi** [comment]
You can just shove monkeys through the chemical-laced tiles till they soak up all the reagents.

**Comment by Iamgoofball** [comment]
@Midaychi valid tactic

**Comment by Lumipharon** [comment]
Speaking of monkeys an chemicals, I'm not sure what the cause, but chemicals dont seem to work on them properly.

I was testing out atropine (oh god, atropine in the air, everyone mute) on a monkey at about -50 health, and it didn't work at all. When I tested it on the other chemist, it worked fine. Also whenI tried to cryo the monkey, it wouldn't heal AT ALL for a good minute or 2, then started healing at a glacially slow speed (healed like, 5 brute over 5 minutes before I just gave up and used bruisepacks). 

Cryo was cold and had chems, and was working fine on people, so I don't know what the problem is, but it makes testing chems a lot more difficult.

**Comment by Iamgoofball** [comment]
@Lumipharon i think someone fucked reagents and monkeys, this has been occuring for awhile. I'll look into it.

**Comment by kevinz000** [comment]
Syptic powder. So much screaming its goon.

**Comment by kevinz000** [comment]
Will the flammable chemicals ... be flammable? (Looking at welder fuel)

**Comment by Marflowerose** [comment]
It might be worth checking out how tiny amount of virus blood is actually required for an infection. Also, this update combined with crank+bath salts+omnizine will result in a very fun Rage Cage.

**Comment by Dorsisdwarf** [comment]
>getting enough drugs to flood the station

File: code/LINDA/LINDA_turf_tile.dm [hunk 1]
```diff
@@ -241,6 +244,14 @@ turf/simulated/proc/share_temperature_mutual_solid(turf/simulated/sharer, conduc
...
 		if("sleeping_agent")
 			overlays.Add(SSair.sleeptoxin_overlay)
+		if("chem_smoke")
+			var/obj/effect/overlay/chem = new /obj/effect/overlay()
+			chem.icon = 'icons/effects/tile_effects.dmi'
+			chem.icon_state = "chem_smoke"
+			chem.layer = FLY_LAYER
+			chem.mouse_opacity = 0
+			chem.color = mix_color_from_reagents(model.gas_reagents.reagent_list)
+			overlays.Add(chem)
```
Comment (Aranclanos) [inline code/LINDA/LINDA_turf_tile.dm]: THE FUCK ARE YOU THINKING, DO YOU EVEN RUN MASS TESTS?

File: code/ATMOSPHERICS/components/unary/vent_pump.dm [hunk 2]
```diff
@@ -269,6 +274,23 @@
 	if (istype(W, /obj/item/weapon/wrench)&& !(stat & NOPOWER) && on)
 		user << "<span class='danger'>You cannot unwrench this [src], turn it off first.</span>"
 		return 1
+	if (istype(W, /obj/item/weapon/reagent_containers))
```
Comment (Aranclanos) [inline code/ATMOSPHERICS/components/unary/vent_pump.dm]: oh yeah hold on let us check each fucking time we are pumping if there are reagents pfhahaha

**Comment by Marflowerose** [comment]
We also need syndicate chemical gas sprinklers for nuclears and traitors now.

**Comment by Aranclanos** [comment]
oh yeah hold on let us check each time we are doing anything if there are reagents in the air with no actual improvements on an already heavy processed sector pfhahaha holy shit are you even trying

**Comment by Stickymayhem** [comment]
Yeah didn't sawu try this and it got closed because it was too laggy?

How did you avoid the same problem?

**Comment by sawu-tg** [comment]
It become overly laggy when someone changed the way reagent transfer work, @Stickymayhem , but @Iamgoofball said he fixed that.

**Comment by Miauw62** [comment]
Codebabbies murdered: 2

What are you going to do next, involve chems in the construction of
machines? Goofdismemberment? Goofmining?
Op 23-mrt.-2015 17:15 schreef "Sawu" <notifications@github.com>:

**Comment by Alek2ander** [comment]
>involve chems in the construction of
machines
Is that a bad thing?

**Comment by Iamgoofball** [comment]
@sawu-tg i fixed it then it unfixed somehow, still trying to refix

**Comment by RemieRichards** [comment]
just don't bother.

I complained to sawu about the multiple overlay objects last time for people a potential source of lag.
and now Aran has popped in and literally flipped his shit at you over the same thing.

**Comment by Iamgoofball** [comment]
i hate atmos

**Pull request closed without merging**